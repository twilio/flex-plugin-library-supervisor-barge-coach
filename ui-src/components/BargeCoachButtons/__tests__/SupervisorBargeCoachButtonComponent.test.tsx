// import * as React from 'react';
// import { render, fireEvent } from '@testing-library/react';
// import { TaskHelper, useFlexSelector, ITask, Manager, IconButton } from '@twilio/flex-ui';
// import { useDispatch, useSelector } from 'react-redux';
// import { AppState, reduxNamespace } from '../../../states'
// import { Actions } from "../../../flex-hooks/states/SupervisorBargeCoach"
// import { Flex, Stack } from "@twilio-paste/core";
// import { isAgentCoachingPanelEnabled } from '../../..';
// import { SyncDoc } from '../../../utils/sync/Sync'
// import { ParticipantTypes } from '@twilio/flex-ui/src/state/Participants/participants.types';


import React from 'react';
import { render } from '@testing-library/react'
import '@testing-library/jest-dom'
import userEvent from '@testing-library/user-event';
import BargeCoachService from '../../../utils/serverless/BargeCoachService';
import { SupervisorBargeCoachButtons } from '../SupervisorBargeCoachButtonComponent';
import { DateTime } from 'luxon';

jest.mock('../../../utils/serverless/BargeCoachService');
jest.mock('react-redux', () => ({
  useSelector: () => ({
    muted: true,
    barge: true,
    enableBargeinButton: true,
    coaching: true,
    enableCoachButton: true,
    privateMode: false
  }),
  useDispatch: () => jest.fn()
}));
jest.mock('../../..', () => ({
  isAgentCoachingPanelEnabled: jest.fn().mockReturnValue(true)
}))

const mockTask = {
  conference: {
    conferenceSid: "CFxxxxxx",
    source:{

    }
  }
}

describe('SupervisorBargeCoachButtons component', () => {

  it('renders correctly', () => {
    const wrapper = render(
      <SupervisorBargeCoachButtons key="callback-component" task={mockTask} />
    )
    expect(wrapper).toMatchSnapshot();
  });
//   it('call button creates outbound call', async () => {
//     const { getByTestId } = render(
//         <SupervisorBargeCoachButtons task={mockTask} />
//     )
//     const callBtn = getByTestId('callbackBtn');
//     expect(callBtn).toBeEnabled();
//     await userEvent.click(callBtn);
//     expect(CallbackService.callCustomerBack).toHaveBeenCalled();
// });
})
