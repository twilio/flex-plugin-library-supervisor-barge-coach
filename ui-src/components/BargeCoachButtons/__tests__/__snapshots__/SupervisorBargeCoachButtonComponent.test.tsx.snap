// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SupervisorBargeCoachButtons component renders correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <style>
      
    table.disable-selection {
      -webkit-touch-callout: none;
      -webkit-user-select: none; 
      -khtml-user-select: none; 
      -moz-user-select: none; 
      -ms-user-select: none; 
      user-select: none;
    }
  
    .disable-selection span::selection{
      background-color: transparent;
    }
    .disable-selection br::selection{
      background-color: transparent;
    }
  
    </style>
    <div>
      <div
        class="css-x2lkna"
        data-paste-element="FLEX"
      >
        <div
          class="css-tfro19"
          data-paste-element="BARGE_COACH_BUTTON_BOX"
        >
          <div
            class="css-ci45q"
            data-paste-element="BARGE_COACH_BUTTON_BOX_CHILD"
          >
            <button
              icon="MuteLargeBold"
              style="width: 44px; height: 44px;"
              title="Unmute"
              variant="secondary"
            />
          </div>
          <div
            class="css-ci45q"
            data-paste-element="BARGE_COACH_BUTTON_BOX_CHILD"
          >
            <button
              disabled=""
              icon="IncomingCallBold"
              style="width: 44px; height: 44px;"
              title="Barge-Out"
              variant="primary"
            />
          </div>
          <div
            class="css-roynbj"
            data-paste-element="BARGE_COACH_BUTTON_BOX_CHILD"
          >
            <button
              icon="DefaultAvatarBold"
              style="width: 44px; height: 44px;"
              title="Disable Coach Mode"
              variant="primary"
            />
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="css-x2lkna"
      data-paste-element="FLEX"
    >
      <div
        class="css-tfro19"
        data-paste-element="BARGE_COACH_BUTTON_BOX"
      >
        <div
          class="css-ci45q"
          data-paste-element="BARGE_COACH_BUTTON_BOX_CHILD"
        >
          <button
            icon="MuteLargeBold"
            style="width: 44px; height: 44px;"
            title="Unmute"
            variant="secondary"
          />
        </div>
        <div
          class="css-ci45q"
          data-paste-element="BARGE_COACH_BUTTON_BOX_CHILD"
        >
          <button
            disabled=""
            icon="IncomingCallBold"
            style="width: 44px; height: 44px;"
            title="Barge-Out"
            variant="primary"
          />
        </div>
        <div
          class="css-roynbj"
          data-paste-element="BARGE_COACH_BUTTON_BOX_CHILD"
        >
          <button
            icon="DefaultAvatarBold"
            style="width: 44px; height: 44px;"
            title="Disable Coach Mode"
            variant="primary"
          />
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
